app/models/concerns/spree/gateway/adyen_gateway.rb

```diff
@@ -64,9 +66,14 @@ def handle_response(response, original_reference = nil)
       )
     end
 
+    def store_for_payment(psp_reference)
```

Comment:
To clarify: I'm not referring to just this method but the whole periphery of it.